opsi/manager/pipeline.py

```diff
@@ -201,13 +345,43 @@ def create_node(self, func: Type[Function], uuid: str):
         and setting node.settings as appropriate
         """
         self.run_order.clear()
-        temp = Node(func, uuid)
+        temp = Node(func, uuid, self.perf_callback)
 
         self.adjList[temp] = set()
         self.nodes[uuid] = temp
 
         return temp
 
+    @property
+    def benchmarking(self):
+        return self._benchmarking
+
+    @benchmarking.setter
+    def benchmarking(self, val):
+        old_benchmarking = getattr(self, "_benchmarking", False)  # handle first run
+        self._benchmarking = bool(val)
+
+        if val:
+            # This is being written with the assumption that the nodes of pipeline will never change
```

Comment:
We need to either disable pipeline import while profiling or cancel the current profile if a new pipeline is imported.

Reply:
Currently pipeline.clear() is called on every import. import_nodetree() calls prune_nodetrees(), which calls clear(), and import_nodetree() calls clear() if a NodeTreeImportError is raised. This is why I added https://github.com/opensight-cv/opensight/blob/45ddf6fb4b94c67eb5bab91bac0490d0a072bda8/opsi/manager/pipeline.py#L396